src/handlers/shared/adapter-middleware.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  SALEOR_API_URL_HEADER,
+  SALEOR_AUTHORIZATION_BEARER_HEADER,
+  SALEOR_DOMAIN_HEADER,
+  SALEOR_EVENT_HEADER,
+  SALEOR_SCHEMA_VERSION,
+  SALEOR_SIGNATURE_HEADER,
+} from "../../const";
+import { createMiddlewareDebug } from "../../middleware/middleware-debug";
+import {
+  HandlerUseCaseResult,
+  HTTPMethod,
+  PlatformAdapterInterface,
+} from "./generic-adapter-use-case-types";
+
+const debug = createMiddlewareDebug("PlatformAdapterMiddleware");
+
+export class PlatformAdapterMiddleware {
+  constructor(private adapter: PlatformAdapterInterface) {}
+
+  withMethod(methods: HTTPMethod[]): HandlerUseCaseResult | null {
+    if (!methods.includes(this.adapter.method)) {
+      return {
+        body: "Method not allowed",
+        bodyType: "string",
+        status: 405,
+      };
+    }
+
+    return null;
+  }
+
+  withSaleorDomainPresent(): HandlerUseCaseResult | null {
```

Comment:
create a todo to check saleorApiUrl. Domain is legacy, we should slowly remove it and check against saleorApiUrl

do not do it in this pr to avoid changes within refactor code